packages/block-library/src/image/edit.js

```diff
@@ -125,6 +213,42 @@ export function ImageEdit( {
 		return pick( getSettings(), [ 'imageDefaultSize', 'mediaUpload' ] );
 	}, [] );
 
+	// A callback passed to MediaUpload,
+	// fired when the media modal closes.
+	function onCloseModal() {
+		if ( isMediaDestroyed( attributes?.id ) ) {
+			setHasImageLoadError( true );
+		}
+	}
+
+	function clearImageAttributes() {
+		setHasImageLoadError( false );
+		setAttributes( {
+			url: undefined,
+			id: undefined,
+		} );
+	}
+
+	/*
+		 Runs an error callback if the image does not load.
+		 If the error callback is triggered, we infer that that image
+		 has been deleted.
+	*/
```

Comment:
Not sure what the accepted conventions in Gutenberg are regarding multiline non-JSDoc comments. Only making this suggestion as the indentation looks odd and stuck out while reviewing.
```suggestion
	/**
	 * Runs an error callback if the image does not load.
	 * If the error callback is triggered, we infer that that image
	 * has been deleted.
	 *
	 * @param {boolean} isReplaced Whether the image has been replaced.
	 */
```